docs/source/onedal/algorithms/ensembles/decision-forest.rst

```diff
@@ -0,0 +1,218 @@
+
+#. New training set generated from the original one by sampling uniformly and with replacement
+   (bootstrapping).
+#. Impurity metric :math:`I` and impurity reduction
+   :math:`I_{\mathrm{reduction}}` for splitting tree's nodes, calculated as follows:
+    - Gini impurity for classification:
+        - without observation weights: :math:`I(D)=1-\sum_{i=1}^{C}{p_i^2},` where :math:`p_i` is the fraction of observations in subset :math:`D` that belong to the :math:`i`-th class.
+        - with observation weights: :math:`I(D)=1-\sum_{i=1}^{C}{p_i^2},` where :math:`p_i` is the weighted fraction of observations in subset :math:`D` that belong to the :math:`i`-th class,
+	  :math:`p_i=(\sum_{d \in \{d \in D | y_{d}=i\}}w_d)/\sum_{d \in D}w_d,` and :math:`w_d` is a weight of observation :math:`d`. 
+    - Mean-Square Error (MSE) for regression: 
+	- without observation weights: :math:`I(D)=\frac{1}{N} \sum_{i=1}^{N}{(y_i - \bar{y})^2},` where :math:`N=|D|` and :math:`\bar{y}=frac{1}{N} \sum_{i=1}^{N}y_i`.
+	- with observation weights: :math:`I(D)=\frac{1}{W(D)} \sum_{i=1}^{N}w_i{(y_i - \bar{y})^2},` where :math:`N=|D|`, :math:`\bar{y}=\sum_{i=1}^{N}w_{i}y_{i},`, :math:`W(D)=\sum_{i=1}^{N}w_{i},` and :math:`w_i` is a weight of observation :math:`i`.
+    - :math:`I_{\mathrm{reduction}}={I} - (N_{\mathrm{left}}/N_{\mathrm{parent}}*I_{left} + N_{\mathrm{right}}/N_{\mathrm{parent}}*I_{\mathrm{right}}),` where :math:`N_{\mathrm{left}}` and :math:`N_{\mathrm{right}}` are the number of observations in the node on the corresponding side of the split.
+
+Due to each tree being fitted to a bootstrapped subset of observations, those that are not used to
+fit a given tree are referred to as out-of-bag (OOB) observations. We can calculate OOB error as the
+average prediction error on each training observation :math:`x_i`, using only the trees that did not have :math:`x_i`
+in their bootstrap observations.
+
+Mean Decrease Impurity (feature importance MDI) metric is calculated as the sum of the impurity
+reduction over all the nodes that are split on that feature (over all trees), averaged by a number
+of trees.
+
+Let :math:`S=(X,Y)` be the set of observations. Given the training parameters, such as the number of trees
+in the forest (:math:`\mathrm{M}`), the fraction of observations used for the training of one tree
+(:math:`\mathrm{observationsPerTreeFraction}`), and the number of features to try as a possible split per
+node (:math:`\mathrm{featuresPerNode}`), the algorithm does the following:
+
+#. For each tree (:math:`1, \ldots, \mathrm{M}`):
+#. Generate a bootstrapped set of observations with :math:`\mathrm{observationsPerTreeFraction} * |S|`
+   elements in it.
+#. Start with the tree whose depth is equal to :math:`0`.
+#. For each leaf node in the tree:
+    - Randomly choose :math:`\mathrm{featuresPerNode}` features to be tried as possible splits.
```

Comment:
`featurePerNode`, I suggest to introduce one letter to refer to `featurePerNode`, otherwise it looks very inconsistent with the rest of math notations.

Reply:
fixed